docs/sync.md

```diff
@@ -0,0 +1,89 @@
+# Syncing within core
+
+Understanding the [data model](data_model.md) is a prerequisite to understanding this document.
+
+Sync is the means by which a client updates their local state after
+
+Design constraints:
+
++ Secure / Private / Trustless -- cannot depend on server to resolve conflicts
++ Frictionless -- cannot depend on user intervention to resolve conflicts
++ Network Independent -- should allow a user to complete as many operations offline as possible
+
+The local stores of state that are relevant to syncing are:
+
++ `FileMetadataRepo` -- list of all the file metadata objects known locally
++ `DocumentRepo` -- store of every document known locally
++ `LocalChangesRepo` -- a description of all the changes that have been made locally
++ `LastSyncedRepo` -- a simple `get/set` value that the sync algorithm uses to help the server filter out events that
```

Comment:
Where is this? Is this the functionality that is currently in the `FileMetadataRepo`?